Rover/mode_auto.cpp

```diff
@@ -57,6 +57,11 @@ void ModeAuto::update()
             IGNORE_RETURN(mis_change_detector.check_for_mission_change());
         }
     } else {
+        if (mission.num_commands() <= 1) {
+            gcs().send_text(MAV_SEVERITY_NOTICE, "No Mission. Can't start AUTO navigation.");
+            mission.stop();
```

Comment:
I'm not sure this will work if the mission is cleared while the vehicle is already executing a mission.  I think it could lead to a "drive away".  I guess it might work if calling "mission.stop()" leads to Rover's ModeAuto::exit_mission() being executed.

Reply:
It cant be cleared if already executing (armed) a mission....have SITL tested this....and while disarmed: load mission, change to AUTO, clear mission... gives the message until mission is loaded or change mode...you can arm, but it does nothing....if armed and you load a mission, it takes off....if you change mode you cant get back into AUTO without a mission loaded...

Reply:
As some discussion, I think we should change this to something like:

if (!armed && mission.num_commands() <= 1) {
    start_stop();
}

.. so let's remove the return and the mission.stop()